documentation/src/main/asciidoc/user_guide/integrations_hibernate_cache.adoc

```diff
@@ -0,0 +1,506 @@
+=== JPA/Hibernate 2L Cache
+
+Hibernate manages a second-level cache where it moves data into and out as a result of operations performed by `Session` or `EntityManager` (JPA).
+The second-level cache is pluggable via an SPI which Infinispan implements.
+This enables Infinispan to be used as second-level cache for Hibernate.
+
+link:https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#caching[Hibernate documentation]
+contains a lot of information about second-level cache, types of caches...etc.
+This chapter focuses on what you need to know to use Infinispan as second-level cache provider with Hibernate.
+
+Applications running in environments where Infinispan is not default cache provider for Hibernate will need to depend on the correct cache provider version.
+The Infinispan cache provider version suitable for your application depends on the Hibernate version in use:
+
+If using Hibernate 5.2 or higher, the Maven coordinates are the following:
+
+[source, XML, indent=0]
+<dependency>
+   <groupId>org.infinispan</groupId>
+   <artifactId>infinispan-hibernate-cache</artifactId>
+   <version>${version.infinispan}</version> <!-- 9.2.0.Final or higher -->
+ </dependency>
+
+If using Hibernate 5.1, the Maven coordinates are the following:
+
+[source, XML, indent=0]
+<dependency>
+   <groupId>org.infinispan</groupId>
+   <artifactId>infinispan-hibernate-cache-v51</artifactId>
+   <version>${version.infinispan}</version> <!-- 9.2.0.Final or higher -->
+</dependency>
+
+NOTE: For earlier Hibernate versions (e.g. Hibernate 5.0 or before), the Infinispan cache provider was shipped by Hibernate.
+This means that the documentation on how to use it and it's Maven coordinates are located in the
+link:https://docs.jboss.org/hibernate/orm/5.0/userguide/html_single/Hibernate_User_Guide.html#caching-provider-infinispan[Hibernate documentation].
+
+Apart from Infinispan specific configuration, it's worth noting that enabling second cache requires some changes to the descriptor file
+(`persistence.xml` for JPA, `hibernate.cfg.xml` for Hibernate or `application.properties` for Spring).
+To use second level cache, you first need to enable the second level cache so that entities and/or collections can be cached:
+
+.Enable second-level cache
+[cols="1,10"]
+|===
+| JPA       | `<property name="hibernate.cache.use_second_level_cache" value="true"/>`
```

Comment:
I'm removing `hibernate.cfg.xml` for now since we don't have a simple tutorial to test it right now and JPA should be the more popular option at this point.